srml/staking/src/lib.rs

```diff
@@ -1178,6 +1213,10 @@ impl<T: Trait> Module<T> {
 
 		// Increment current era.
 		let current_era = CurrentEra::mutate(|s| { *s += 1; *s });
+
+		// prune journal for last era.
+		<EraSlashJournal<T>>::remove(current_era - 1);
```

Comment:
Couldn't we prune era's with some offset (`X`) and accept reports for last `X` eras?

Reply:
We could, but it wouldn't be of any use with this runtime. Historical slashing just isn't doable without more work beyond this milestone.